taichi/transforms/auto_diff.cpp

```diff
@@ -606,21 +712,28 @@ class MakeAdjoint : public IRVisitor {
       new_if->set_true_statements(std::make_unique<Block>());
       auto old_current_block = current_block;
 
+      forward_backup = if_stmt->true_statements.get();
```

Comment:
There need to be a comment describing what is `forward_backup` and when/where it might be updated/restored.  :P

Reply:
Have added more comments for it.